2017-04-28  Kaiyuan Chen  <classhen@lnxsrv07.seas.ucla.edu>

	* lib/timespec.h:
	Use comparison rather than substraction to compare two time and to avoid time overflow 


--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2017-04-27 23:58:08.235649000 -0700
@@ -45,8 +45,8 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  return  a.tv_sec > b.tv_sec ? 1 : (a.tv_sec < b.tv_sec ?
+				     -1 : a.tv_nsec - b.tv_nsec);
 }
 
 # if ! HAVE_DECL_NANOSLEEP
